src/csharp/Microsoft.Spark/Utils/DependencyProviderUtils.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+
+namespace Microsoft.Spark.Utils
+{
+    internal class DependencyProviderUtils
```

Comment:
any reason this is under `Microsoft.Spark` and `DependencyProvider` is under `Microsoft.Spark.Worker`? I guess it's coming in extension change? If so, should we just put them into one place?

Reply:
Since the Metadata is used by the extension I put `DependencyProviderUtils` in `Microsoft.Spark`.  I only put `DependencyProvider` in `Microsoft.Spark.Worker` because the worker is the only consumer of the class.  But if it makes it easier I can move everything into `DependencyProvider` and put it in `Microsoft.Spark`